test/with_dummyserver/test_chunked_transfer.py

```diff
@@ -33,9 +36,9 @@ def socket_handler(listener):
 
         self._start_server(socket_handler)
 
-    def test_chunks(self):
+    def test_chunks(self) -> None:
         self.start_chunked_handler()
-        chunks = ["foo", "bar", "", "bazzzzzzzzzzzzzzzzzzzzzz"]
+        chunks = [b"foo", b"bar", b"", b"bazzzzzzzzzzzzzzzzzzzzzz"]
```

Comment:
It does seem like `request_chunked` explicitly supports `str` chunks:

https://github.com/urllib3/urllib3/blob/cff8b5ba3d2ba9338e31c102e1e91273124df297/src/urllib3/connection.py#L326-L327

I'm not sure if this is a case "we support it but don't do this, so type annotation forbid this" or "this is supported so `_TYPE_BODY` should allow it". To me chunked + str sounds like a bad idea but I don't really know.

Reply:
@sethmlarson @pquentin Could you please elaborate on this?

Reply:
I'd lean for "we support it but don't do this, so type annotation forbid this".

@sethmlarson thoughts?

Reply:
++ on support with type annotations forbidding it.